common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs

```diff
@@ -228,6 +265,47 @@ public sealed class AddressRange {
             _cur = 0;
         }
 
+        /// <summary>
+        /// Convert address range to cidr formatted ip strings
+        /// </summary>
+        /// <param name="sb"></param>
+        /// <returns></returns>
+        private void AppendTo(StringBuilder sb) {
+            long start = Low;
+            long end = High;
+            var first = true;
+            while (end >= start) {
+                byte maxSize = 32;
+                while (maxSize > 0) {
+                    var mask = (long)(Math.Pow(2, 32) - Math.Pow(2, 32 - (maxSize - 1)));
+                    var maskBase = start & mask;
+                    if (maskBase != start) {
+                        break;
+                    }
+                    maxSize--;
+                }
+                var x = Math.Log(end - start + 1) / Math.Log(2);
```

Comment:
>x = Math.Log(end - start + 1) / Math.Log(2); [](start = 20, length = 44)

I bet there is a good way to this without log..

Reply:
Wont fix for now